File: generator/src/main/groovy/ch/bailu/gtk/tag/MemberTag.java
package ch.bailu.gtk.tag;

public class MemberTag extends NamedTag {
    private String value = null;

    public MemberTag(Tag parent) {
        super(parent);
    }


    @Override
    public void setAttribute(String name, String value) {
        if ("value".equals(name)) {
            this.value = value;
        } else {
            super.setAttribute(name, value);
        }
    }

    public String getValue() {
        return value;
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/parser/tag/MemberTag.kt
package ch.bailu.gtk.parser.tag

class MemberTag(parent: TagWithParent) : ParameterTag(parent) {

    override fun getType(): String {
        return if ("" == super.getType()) "int" else super.getType()
    }

    override fun getName(): String {
        return super.getName().uppercase()
    }
}
